internal/provider/data_source_oauth_client.go

```diff
@@ -132,6 +141,7 @@ func dataSourceTFEOAuthClientRead(d *schema.ResourceData, meta interface{}) erro
 	}
 	d.Set("service_provider", oc.ServiceProvider)
 	d.Set("service_provider_display_name", oc.ServiceProviderName)
+	d.Set("organization_scoped", oc.OrganizationScoped)
```

Comment:
Will this behave correctly on TFE where this key is missing/false (This would be the default empty value when the key) ?

[The resource](https://github.com/hashicorp/terraform-provider-tfe/blob/main/internal/provider/resource_tfe_oauth_client.go#L119) has the benefit of having a default true value.

But it would seem like you need to invert it if the field is fully missing. Unfortunately, your go-tfe field is not a pointer so it's not easy to know whether the field was missing or literally false.

What are your thoughts?

Reply:
In atlas, organization_scoped boolean DEFAULT true NOT NULL is set as a restriction. So then the missing case shouldn't be possible correct.

Reply:
I was talking about when you are using this against TFE that haven't added this API yet and therefore organization-scoped would be missing from the response altogether

Reply:
I see two options:

1. Release a known 'incorrect' value for TFE customers until they upgrade to latest release and get the API
2. Change go-tfe beta field (this is why they are beta!) to be a pointer that can be nil. The nil value will indicate the API is missing.
3. Change the API altogether to be "project-scoped", which makes missing/default/false values all alike. (This is obviously a crazy choice at this stage)

Reply:
Which one would you suggest? I would have to confirm with my PM if option 1 is ok to do.

Reply:
I think the best compromise is to change the field to be a pointer attribute in the underlying go-tfe interface. It's hard to justify releasing a known bug when it can be fixed relatively easily

Reply:
I think I am going to go ahead and implement #2, that one seems like the best solution in this case.